api/v1/services/EventService.js

```diff
@@ -0,0 +1,153 @@
+var Checkit = require('checkit');
+var _Promise = require('bluebird');
+var _ = require('lodash');
+var moment = require('moment');
+
+var Event = require('../models/Event');
+var Location = require('../models/Location');
+var LocationEvents = require('../models/LocationEvents');
+
+var errors = require('../errors');
+var utils = require('../utils');
+
+/**
+ * Creates a new event and locations with the specified parameters. Validation is performed on-save only
+ * @param  {String}         name            The event's name
+ * @param  {String}         short_name            The event's name
+ * @param  {String}         description     The event's description
+ * @param  {Number|Boolean} qr_code         Whether the event needs a QR code or not
+ * @param  {Number}         start_time      Time at which the event starts
+ * @param  {Number}         end_time        Time at which the event ends
+ * @param  {String}         eventTag        Array of Location objects, or id referencing them.
+ * @param  {[Object]}       locations       Array of Location objects, or id referencing them.
+ *                                          Location object containing three parameters:
+ *                                              name:      name of the location,
+ *                                              latitude:  latitude of the location,
+ *                                              longitude: longitude of the location
+ *                                          Otherwise, we expect just an id.
+ */
+module.exports.createEvent = function (name, short_name,
```

Comment:
Like I said earlier, Bookshelf allows you to create an instance of a model by passing in a JS object of attributes